tests/test_widget.py

```diff
@@ -317,3 +316,76 @@ def check_dialog_and_click_no(dialog):
         dialog_action=check_dialog_and_click_no,
         time_out=5,
     )
+
+
+def test_radius_save_preserves_old_data(cpt_widget):
+    """Test saving a new particle radius does not mess up the previous one."""
+    # setup measurement layer
+    # layer_measurements = cpt_widget._setup_measurement_layer()
+    layer_measurements = cpt_widget.viewer.add_points(name="Radii and Lengths", ndim=2)
+
+    # add a new particle and calculate radius
+    cpt_widget.particle_decays_menu.setCurrentIndex(1)
+    # layer_measurements.add([[0, 0, 0, 1], [0, 0, 1, 0], [0, 0, 0, -1]]) # 1px radius
+    layer_measurements.add([[0, 1], [1, 0], [0, -1]])  # 1px radius
+    layer_measurements.selected_data = {0, 1, 2}
+    cpt_widget._on_click_radius()
```

Comment:
Here it's actually not so clear so, unfortunately, I need some hand-holding comments...
```suggestion
    # Add first particle
    cpt_widget.particle_decays_menu.setCurrentIndex(1) # adds new row
    # Add three points on a circle of radius 1 px
    # layer_measurements.add([[0, 0, 0, 1], [0, 0, 1, 0], [0, 0, 0, -1]]) # 1px radius
    layer_measurements.add([[0, 1], [1, 0], [0, -1]])  # 1px radius
    # Select the first three points in the layer
    layer_measurements.selected_data = {0, 1, 2}
    # User clicks the radius calculation button
    cpt_widget._on_click_radius()
```